perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_member_attribute_def_virt_loa.java

```diff
@@ -22,8 +22,8 @@
 public class urn_perun_member_attribute_def_virt_loa extends MemberVirtualAttributesModuleAbstract implements MemberVirtualAttributesModuleImplApi {
 
 	@Override
-	public void checkAttributeSemantics(PerunSessionImpl sess, Member member, Attribute attribute) throws InternalErrorException, WrongAttributeValueException {
-		if(!attribute.equals(getAttributeValue(sess, member, attribute))) throw new WrongAttributeValueException(attribute, member, "Attribute value is not the highest value from member's UserExtSources Loas.");
+	public void checkAttributeSemantics(PerunSessionImpl sess, Member member, Attribute attribute) throws InternalErrorException, WrongReferenceAttributeValueException {
+		if(!attribute.equals(getAttributeValue(sess, member, attribute))) throw new WrongReferenceAttributeValueException(attribute, getAttributeValue(sess, member, attribute), member, null, "Attribute value is not the highest value from member's UserExtSources Loas.");
```

Comment:
Prevent duplicity of this call `getAttributeValue(sess, member, attribute).` You should do something like:
`
Attribute currentAttribute = getAttributeValue(sess, member, attribute);
if(!attribute.equals(currentAttribute)) throw new WrongReferenceAttributeValueException(attribute, currentAttribute, member, member, "message");
`

Reply:
Okay, there should not be this duplicity now, thank you.